source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -321,15 +321,28 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
 
             if (fallbackElementRenderer)
             {
-                fallbackControl = fallbackElementRenderer.Render(fallbackElement, renderContext, renderArgs);
-
-                renderedElement = fallbackElement;
+                try
+                {
+                    fallbackControl = fallbackElementRenderer.Render(fallbackElement, renderContext, renderArgs);
+                    renderedElement = fallbackElement;
+                }
+                catch (winrt::hresult_error const& ex)
+                {
+                    if (ex.code() == E_PERFORM_FALLBACK)
+                    {
+                        std::tie(fallbackControl, renderedElement) = RenderFallback(fallbackElement, renderContext, renderArgs);
+                    }
+                    else
+                    {
+                        throw(ex);
+                    }
+                }
```

Comment:
Should keep the `fallbackControl` nullcheck that triggers the recursive call to `RenderFallback` here as well?